### PR TITLE
Docs/lakefs setup with dynamodb

### LINKED ISSUE #3895: KV Docs: lakeFS setup with DynamoDB
- Docs base PR: https://github.com/treeverse/lakeFS/pull/3878

Create a PR over the base PR and add page on KV setup with DynamoDB

### PR DESCRIPTION
**Pull request by itai-david**
Closes #3895 

First go - Updated lakeFS deployement on AWS with DynamoDB

### CODE REVIEW

**Comment by itai-david** [comment]
closes #3895

File: docs/deploy/aws.md [hunk 1]
```diff
@@ -19,8 +19,16 @@ Expected deployment time: 25 min
 
 {% include_relative includes/prerequisites.md %}
 
-## Creating the Database on AWS RDS
-lakeFS requires a PostgreSQL database to synchronize actions on your repositories.
+## Preparing the Database for the Key Value Store
+lakeFS uses a key-value store to synchronize actions on your repositories. Out of the box, this key-value store can rely on DynamoDB or a PostgreSQL DB. As lakeFS open source, you can also write your own implementation and use any other DB
+The following two sections explain how to setup either DynamoDB or PostgreSQL as key-value backing DB
+
+### DynamoDB on AWS
```
Comment (itaiad200) [inline docs/deploy/aws.md:26]: I think we should put this second as it is less battle proven than postgres.

File: docs/deploy/aws.md [hunk 2]
```diff
@@ -36,12 +44,36 @@ If you already have a database, take note of the connection string and skip to t
...
    ```yaml
    ---
    database:
-     connection_string: "[DATABASE_CONNECTION_STRING]"
+     type: "dynamodb"
+     dynamodb: 
+       table_name: "[DYNAMODB_TABLE_NAME]"
+       aws_region: "[DYNAMODB_AWS_REGION]"
+       aws_access_key_id: "[AWS_ACCESS_KEY_ID]"
+       aws_secret_access_key: "[AWS_SECRET_ACCESS_KEY]"
```
Comment (itaiad200) [inline docs/deploy/aws.md:58]: We don't need these 3, just like you don't provide them for the S3 blockstore.

File: docs/deploy/aws.md [hunk 3]
```diff
@@ -60,14 +92,30 @@ If you already have a database, take note of the connection string and skip to t
...
-command to demonstrate starting lakeFS using Docker:
-
+To support container-based environments like AWS ECS, lakeFS can be configured using environment variables. Here are a couple of `docker run` 
+commands to demonstrate starting lakeFS using Docker:
+#### With PostgreSQL
+```sh
+docker run \
+  --name lakefs \
+  -p 8000:8000 \
+  -e LAKEFS_DATABASE_KV_ENABLED="true"
```
Comment (itaiad200) [inline docs/deploy/aws.md:102]: This won't be needed by the time it's merged.

File: docs/deploy/aws.md [hunk 4]
```diff
@@ -60,14 +92,30 @@ If you already have a database, take note of the connection string and skip to t
...
+  -e LAKEFS_DATABASE_TYPE: "dynamodb" \
+  -e LAKEFS_DATABASE_DYNAMODB_AWS_ACCESS_KEY_ID="[AWS_ACCESS_KEY_ID]" \
+  -e LAKEFS_DATABASE_DYNAMODB_AWS_SECRET_ACCESS_KEY="[AWS_SECRET_ACCESS_KEY]" \
+  -e LAKEFS_DATABASE_DYNAMODB_AWS_REGION="[DYNAMODB_AWS_REGION]" \
```
Comment (itaiad200) [inline docs/deploy/aws.md:106]: Same here.

File: docs/deploy/aws.md [hunk 5]
```diff
@@ -60,14 +92,30 @@ If you already have a database, take note of the connection string and skip to t
...
-
+To support container-based environments like AWS ECS, lakeFS can be configured using environment variables. Here are a couple of `docker run` 
+commands to demonstrate starting lakeFS using Docker:
+#### With PostgreSQL
+```sh
+docker run \
+  --name lakefs \
+  -p 8000:8000 \
+  -e LAKEFS_DATABASE_KV_ENABLED="true"
+  -e LAKEFS_DATABASE_TYPE: "dynamodb" \
```
Comment (itaiad200) [inline docs/deploy/aws.md:103]: Where's the table name? That one is actually important.

File: docs/deploy/aws.md [hunk 6]
```diff
@@ -60,14 +92,30 @@ If you already have a database, take note of the connection string and skip to t
...
+  -e LAKEFS_BLOCKSTORE_TYPE="s3" \
+  treeverse/lakefs:latest run
+```
+#### With PostgreSQL
 ```sh
 docker run \
   --name lakefs \
   -p 8000:8000 \
-  -e LAKEFS_DATABASE_CONNECTION_STRING="[DATABASE_CONNECTION_STRING]" \
+  -e LAKEFS_DATABASE_KV_ENABLED="true"
```
Comment (itaiad200) [inline docs/deploy/aws.md:116]: Unnecessary

File: docs/reference/database-migration.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+has_children: false
+---
+
+# lakeFS Database Migrate
+{: .no_toc }
+
+**Note:** Feature in development
+{: .note }
+
+The lakeFS database migration tool simplifies switching from one database implementation to another.
```
Comment (itaiad200) [inline docs/reference/database-migration.md:16]: I would add an example to make it clear.
```suggestion
The lakeFS database migration tool simplifies switching from one database implementation to another, e.g. from Postgres to DynamoDB.
```

**Review by itaiad200 — CHANGES_REQUESTED** [review]
See comments inline. I don't think we should merge it to master, it also includes many irrelevant commits.

**Review by N-o-Z — CHANGES_REQUESTED** [review]
@itai-david As described in the original issue (https://github.com/treeverse/lakeFS/issues/3895)
This PR should be to https://github.com/treeverse/lakeFS/pull/3878 and not master

File: docs/deploy/aws.md (hunk 1 above)
Comment (itai-david) [reply docs/deploy/aws.md]: Makes total sense. Also switched the order of the configuration/docker snippets to reflect the same order

File: docs/deploy/aws.md (hunk 2 above)
Comment (itai-david) [reply docs/deploy/aws.md]: Removed

File: docs/deploy/aws.md (hunk 3 above)
Comment (itai-david) [reply docs/deploy/aws.md]: Correct - removed

File: docs/deploy/aws.md (hunk 5 above)
Comment (itai-david) [reply docs/deploy/aws.md]: Added.
Although, we do have a default table name, which is workable

File: docs/deploy/aws.md (hunk 4 above)
Comment (itai-david) [reply docs/deploy/aws.md]: Removed

File: docs/deploy/aws.md (hunk 6 above)
Comment (itai-david) [reply docs/deploy/aws.md]: Removed

File: docs/reference/database-migration.md (hunk 7 above)
Comment (itai-david) [reply docs/reference/database-migration.md]: This one is coming from the migration PR. It is no longer part of this PR as I changed the base

**Comment by itai-david** [comment]
> See comments inline. I don't think we should merge it to master, it also includes many irrelevant commits.

@itaiad - merge target branch modified to reflect the original base. Thanks for pointing this out

File: docs/deploy/aws.md [hunk 8]
```diff
@@ -19,8 +19,11 @@ Expected deployment time: 25 min
 
 {% include_relative includes/prerequisites.md %}
 
-## Creating the Database on AWS RDS
-lakeFS requires a PostgreSQL database to synchronize actions on your repositories.
+## Preparing the Database for the Key Value Store
+lakeFS uses a key-value store to synchronize actions on your repositories. Out of the box, this key-value store can rely on DynamoDB or a PostgreSQL DB. As lakeFS open source, you can also write your own implementation and use any other DB
```
Comment (N-o-Z) [inline docs/deploy/aws.md:23]: Maybe generalize it more and not mention the current supported drivers explicitly, so we won't have to maintain this documentation everytime we add a new driver

**Review by N-o-Z — APPROVED** [review]
Looks good! One suggestion

File: docs/deploy/aws.md (hunk 8 above)
Comment (itai-david) [reply docs/deploy/aws.md:23]: I guess that if we do add other drivers, we will need to update this doc one way or another, and at least add another configuration section, similar to the 2 we have below.
I think it is best to be as explicit as possible with manuals, so I would vote to leave the specification and expand it in the future (if needed). WDYT?

**Review by itaiad200 — APPROVED** [review]

**Pull request merged**